### PR TITLE
chore: fixtures-via-plugin implementation

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

**Review by yury-s — APPROVED** [review]
Looks reasonable, would be nice if Ross had a look at it too.

File: utils/generate_types/overrides-test.d.ts [hunk 1]
```diff
@@ -56,9 +56,14 @@ export interface FullProject<TestArgs = {}, WorkerArgs = {}> {
 
 type LiteralUnion<T extends U, U = string> = T | (U & { zz_IGNORE_ME?: never });
 
+export interface TestPlugin {
+  fixtures?: Fixtures;
```
Comment (rwoll) [inline utils/generate_types/overrides-test.d.ts:60]:
```suggestion
  fixtures?: Fixtures;
  // [internal] NB: other fields are generated from the docs
```

File: tests/playwright-test/plugins.spec.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      import plugin from './plugin.ts';
+      module.exports = {
+        plugins: [ plugin ],
+      };
+    `,
+    'plugin.ts': `
+      export default {
+        fixtures: {
+          foo: 42,
+          myBrowserName: async ({ browserName }, use) => { await use(browserName) }
```
Comment (rwoll) [inline tests/playwright-test/plugins.spec.ts:131]: We don't need to do this now, but we should make sure the various fixture options work and/or still validate nicely.

File: packages/playwright-test/src/reporters/raw.ts [hunk 3]
```diff
@@ -317,4 +317,12 @@ function dedupeSteps(steps: JsonTestStep[]): JsonTestStep[] {
   return result;
 }
 
+function filterOutPrivateFields(object: any): any {
+  if (!object || typeof object !== 'object')
+    return object;
+  if (Array.isArray(object))
+    return object.map(filterOutPrivateFields);
+  return Object.fromEntries(Object.entries(object).filter(entry => !entry[0].startsWith('_')).map(entry => [entry[0], filterOutPrivateFields(entry[1])]));
+}
```
Comment (rwoll) [inline packages/playwright-test/src/reporters/raw.ts:326]: Let's make sure this change lands regardless of the fate of plugins.

**Review by rwoll — APPROVED** [review]

File: packages/playwright-test/src/reporters/raw.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright-test/src/reporters/raw.ts:326]: I think we'll just _plugins

**Pull request merged**